File: apps/blueprint-gym/src/Domain/ExerciseTracker/Models/Exercise.cs
using System.Collections.Generic;
using BlueprintGym.Domain.Constants.Enums;
using BlueprintGym.Domain.Core.Cosmos;
using BlueprintGym.Domain.Core.Interfaces;
using BlueprintGym.Domain.Core.Models;

namespace BlueprintGym.Domain.ExerciseTracker.Models
{
  public class Exercise : CosmosEntity, ICosmosEntity
  {
    public override string PK => $"{this.GetType().Name}-{this.ExerciseId}";
    public string ExerciseId => this.Id;
    public string ExerciseName { get; set; }
    public string Description { get; set; }
    public IEnumerable<string> MusclesWorked { get; set; }
    public IEnumerable<string> ExerciseLabels { get; set; }
    public ExerciseState State { get; set; }
    public MuscleSpecificity MuscleSpecificity { get; set; }
    public FitnessDifficulty Difficulty { get; set; }
    public MemberLink Author { get; set; }
    public ExerciseLink ParentVariationExercise { get; set; }
    public MuscleSpecificity PreviousMuscleSpecificity { get; set; }
    public string VideoUrl { get; set; }
    public BlueScoreFormula Formula { get; set; }
  }
}

File: apps/blueprint-gym/exercise-tracker/models/exercise.go
package models

import (
	"../constants"
	"go.mongodb.org/mongo-driver/bson/primitive"
)

type AllExercises struct {
	ID           primitive.ObjectID `json:"_id,omitempty" bson:"_id,omitempty"`
	ExerciseRefs []ExerciseRef
}

type Exercise struct {
	ID                      primitive.ObjectID `json:"_id,omitempty" bson:"_id,omitempty"`
	ExerciseName            string
	Description             string
	MusclesWorked           []string
	Labels                  []string
	State                   constants.ExerciseState
	Difficulty              constants.FitnessDifficulty
	Author                  MemberLink
	ParentVariationExercise ExerciseLink
}

type ExerciseRef struct {
	ExerciseId         string
	ExerciseName       string
	DescriptionSnippet string
	MusclesWorked      []string
	State              constants.ExerciseState
	Difficulty         constants.FitnessDifficulty
}
